app/scripts/controllers/landingPage.js

```diff
@@ -111,6 +111,30 @@ angular.module('openshiftConsole')
     });
 
     function dataLoaded() {
+      // Check for a service class param to launch the catalog flow for
+      var paramClass = $location.search()['serviceClass'];
+      if (paramClass) {
+        // Lookup the class by class name e.g. cakephp-mysql-persistent
+        var paramItem = _.find($scope.catalogItems, {
+          name: paramClass
+        });
+        // Lookup the class by display name e.g. "CakePHP + MySQL (Persistent)"
```

Comment:
It looks like you might have the order backwards here, checking display name first above and then `metadata.name` here.